R/crunchbox.R

```diff
@@ -41,6 +46,28 @@ crunchBox <- function (dataset, filters=crunch::filters(dataset), ...) {
     ## Add "where" after so that it no-ops if variablesFilter returns NULL (i.e. no filter)
     payload$where <- variablesFilter(dataset)
 
+    ## Add colors if they exist to the payload
+    if (!missing(brand_colors)) {
+        if (!is.list(brand_colors) ||
+            any(!names(brand_colors) %in% c("primary", "secondary", "message"))) {
+            halt(sQuote("brand_colors"), " must be a named list with only ",
+                 serialPaste(dQuote(c("primary", "secondary", "message")), collapse = "or"))
+        }
+        payload$display_settings$palette$brand_colors <- brand_colors
+    }
+    if (!missing(static_colors)) {
+        if (!is.list(static_colors)) {
+            halt(sQuote("static_colors"), " must be a list of characters")
```

Comment:
Would a character vector be okay here? If so I think that's probably easier than a list. Also the error specifies that it be a list of character strings, but I don't think that's actually being checked.